opentelemetry-metrics-exporters/src/main/java/com/cumulocity/exporters/platform/PlatformMetricExporter.java

```diff
@@ -0,0 +1,107 @@
+package com.cumulocity.exporters.platform;
+
+import com.cumulocity.rest.representation.application.microservice.MicroserviceBillingRepresentation;
+import com.cumulocity.sdk.client.RestConnector;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.DoublePointData;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import lombok.extern.slf4j.Slf4j;
+import org.joda.time.DateTime;
+import org.springframework.context.ApplicationEventPublisher;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static io.opentelemetry.sdk.metrics.data.MetricDataType.*;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+
+
+@Slf4j
+public class PlatformMetricExporter implements MetricExporter {
+
+
+    private RestConnector restConnector;
+
+    private ApplicationEventPublisher applicationEventPublisher;
+
+    public PlatformMetricExporter(RestConnector restConnector, ApplicationEventPublisher applicationEventPublisher) {
+        this.restConnector = restConnector;
+        this.applicationEventPublisher = applicationEventPublisher;
+    }
+
+    private void exportNonGaugeMetrics(MetricData metric) {
+        String name = metric.getName();
+        double value = 0;
+        if (metric.getType().equals(LONG_SUM)) {
+            value = metric.getLongSumData().getPoints().stream().findFirst().get().getValue();
+        } else if (metric.getType().equals(DOUBLE_SUM)) {
+            value = metric.getDoubleSumData().getPoints().stream().findFirst().get().getValue();
+        } else {
+            value = 0;
+        }
+        log.info("Sending metric for tenant {} named {} whose value is {} and whose type is {}", restConnector.getPlatformParameters().getTenantId(), name, value, metric.getType().name());
+        MicroserviceBillingRepresentation representation = new MicroserviceBillingRepresentation();
+        representation.setDateTime(DateTime.now());
+        representation.set(value, name);
+        restConnector.post("/application/currentApplication/billing",
+                APPLICATION_JSON_TYPE, representation);
+    }
+
+    private void exportGaugeMetrics(MetricData metric) {
+        Collection<DoublePointData> points = metric.getDoubleGaugeData().getPoints();
```

Comment:
Limitation of GaugeMetric:

Instrument name - Loriot Gauge Metrics

send 3 metrics in a window: 100 12:30 (attribute 12:30-12:30), 200 12:31(attribute 12:31-12:31), 300 12:32(attribute 12:32-12:32)

These are 3 points in the instrument loriot Gauge metrics

3 points = 3 callbacks

point data
100   12:30
200   12:31
300   12:32

The attributes are the way in which we can record the time at which the metric was actually instrumented. This logic creates a PointData-Time Map, using which we can get the most recent Metric sent and export that. If we do not send attributes in the metric points, only the first point is recorded, as the others are discarded due to the previous point already being recorded